crates/xtask/src/main.rs

```diff
@@ -1,18 +1,76 @@
-use clap::Parser;
+use block_oracle::config::{Config, ConfigFile};
+use clap::Parser as _;
+use std::path::PathBuf;
 
+mod contracts;
 mod message_samples;
 
 /// Block Oracle automation scripts
-#[derive(Parser)]
-#[clap()]
+#[derive(clap::Parser)]
 enum Tasks {
     /// Compile and display encoded message samples
     EncodeMessageSamples,
+    /// Queries the Epoch Manager for the current epoch
+    CurrentEpoch {
+        #[clap(short, long, value_enum)]
+        environment: Environment,
+    },
+    /// Sends a message to the DataEdge contract
+    SendMessage {
+        #[clap(short, long, value_enum)]
+        environment: Environment,
+        #[clap(value_enum)]
+        message: Message,
+    },
 }
 
-fn main() -> anyhow::Result<()> {
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    use Tasks::*;
     match Tasks::parse() {
-        Tasks::EncodeMessageSamples => message_samples::encode()?,
+        EncodeMessageSamples => message_samples::encode()?,
+        CurrentEpoch { environment } => contracts::current_epoch(environment).await?,
+        SendMessage {
+            environment,
+            message,
+        } => contracts::send_message(message, environment).await?,
     };
     Ok(())
 }
+
+#[derive(clap::ValueEnum, Clone)]
+pub enum Message {
+    Reset,
+}
```

Comment:
This `enum` is effectively duplicating a subset of the `oracle-encoder` logic. We can get more features with less code if we simply ask the user for a `.json` file and compile that into a payload on the fly using `oracle-encoder` as a library:

`$ cargo xtask send-message ../config/dev/config.toml reset-message.json`

Reply:
Referenced in a new issue: https://github.com/graphprotocol/block-oracle/issues/157